message.go

```diff
@@ -915,7 +1016,15 @@ func (m *Message) unmarshalField(field reflect.Value, rv reflect.Value) error {
 			return fmt.Errorf("%v: %v", errUnmarshalNonMessage, rv.Type())
 		}
 
-		return msg.unmarshal(field.Interface())
+		fp := field
+		if field.IsNil() {
```

Comment:
Why exactly is this needed? Is this an old issue or due to your changes?

Reply:
Let me find the test case that required this and I will add it to the tests.

Reply:
I will open a separate request about this. This is for automatically initializing pointers to nested structs.